jwk/jwks.go

```diff
@@ -0,0 +1,22 @@
+package jwk
+
+import "fmt"
+
+type JWKSet struct {
+	Keys   []JWK `json:"keys"`
+	keyMap map[string]*JWK
+}
+
+func (j *JWKSet) Get(kid string) (*JWK, error) {
+	if j.keyMap == nil {
+		j.keyMap = make(map[string]*JWK)
```

Comment:
Race condition. You really need to be using a mutex here, you never know how your caller is going to be using this.

This is also actually a good case for using a constructor. Most times, no, but if you’re creating a map backing store for faster look up, then just make the type `map[string]*JWK` and have a `NewJWKSet(...string)` that converts the slice to a map.

Reply:
Problem is that `JWKSet` is also being served as a `JSON` object as the `/jwks.json` endpoint

Reply:
You can implement a `MarshalJSON` that will create an ordered list and then render that.

This also prevents the situation of the two possibly becoming out of sync. i.e. if I were to mess with `jwks.Keys = append(jwks.Keys, …)` the map won’t update, right?

Reply:
I removed the premature optimisation as JWKS is expected to be small (And  I know it is as it's a controlled value by my side)